Reorganize pulse visualization code

mtreinish opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary

An improved pulse visualizer was added in #2213, but in that commit the
additional matplotlib logic and utility functions were added on to
existing modules that used matplotlib. This makes maintaining the code
more difficult for 2 reasons, first the pulse visualization code is
mostly self contained and didn't really have anything to with the
modules it was added onto, and secondly the existing matplotlib code
was already quite large. This commit reorganizes the pulse visualization
code to be easier to maintain by moving all pulse related lib code into
separate modules in a pulse/ directory.

### Details and comments

mtreinish commented:
> Thank you for tackling this @mtreinish. If we are moving pulse visualization into its own module, should we also not move circuit visualization into its own module?

I thought of doing that as it would clean up the visualization directory a lot. But, since we've declared the circuit visualization interfaces as stable I was concerned about breaking users who could be using them in their current location.

taalexander commented:
Can we change the imports to make it continue to work, with the circuit visualization in the new module?

mtreinish commented:
> Can we change the imports to make it continue to work, with the circuit visualization in the new module?

That wasn't actually what I was talking about, the modules' location and import path is part of a python library's api. If we change that it potentially could break external users. Even if we update import paths in `qiskit.visualization.__init__` to use the functions from a new location people could be using things from individual backend modules and that's still a breaking change. To do it properly we'd have to add compat shim modules (like what we did for `qiskit._util` in 0.8) and deprecate them before we can remove them. This feels like a lot of effort just for a code reorganization that doesn't really provide a concrete benefit besides making it look a little cleaner. But, regardless of whether we move the circuit's visualization modules around or not that change is outside the scope of this PR because it would not be included in this.

mtreinish commented on qiskit/visualization/pulse/qcstyle.py:
```diff
@@ -0,0 +1,86 @@
+# pylint: disable=missing-docstring
+
+
+class OPStyleSched:
```
Normally I like to keep one logical change per PR, just because it makes it easier to review and revert/bisect if something goes wrong. But I think since we have to update all the imports anyway this is fine. I'll push an update for this shortly

mtreinish commented:
Ok i updated the docstrings in: https://github.com/Qiskit/qiskit-terra/pull/2340/commits/d5a518669066b86d67021bf7307595d89ad0ba05 and I also renamed the updated class names instead of just doing a s/OP/Pulse/g which resulted in some weird names :)
